src/TufValidatedComposerRepository.php

```diff
@@ -47,19 +49,38 @@ public function __construct(array $repoConfig, IOInterface $io, Config $config,
             $fs = new Filesystem();
             $fs->ensureDirectoryExists($repoPath);
 
-            // We expect the repository to have a root metadata file in a known
-            // good state. Copy that file to our persistent storage location if
-            // it doesn't already exist.
-            $rootFile = $repoPath . '/root.json';
+            // For testing purposes, allow the file fetcher to be passed in the repository configuration.
+            $fetcher = $repoConfig['tuf']['_fileFetcher'] ?? GuzzleFileFetcher::createFromUri($url);
+            $this->updater = new Updater($fetcher, [], new FileStorage($repoPath));
+
+            // If we don't yet have up-to-date TUF metadata in place, download the root
+            // data from the server and validate it against the hash(es) in the repository
+            // configuration.
+            $rootFile = $repoPath . DIRECTORY_SEPARATOR . 'root.json';
             if (!file_exists($rootFile)) {
-                $sourcePath = realpath($repoConfig['tuf']['root']);
-                if (!$fs->copy($sourcePath, $rootFile)) {
-                    throw new FilesystemException("Could not copy '$sourcePath' to '$rootFile");
-                }
-            }
+                $rootInfo = $repoConfig['tuf']['root'];
 
-            $fetcher = GuzzleFileFetcher::createFromUri($url);
-            $this->updater = new Updater($fetcher, [], new FileStorage($repoPath));
+                $fetcher->fetchMetadata('root.json', $rootInfo['length'])
+                    ->then(function (StreamInterface $stream) use ($rootFile, $rootInfo) {
```

Comment:
It difficult with these very similarly-named variables to keep them straight. Let's rename them to clarify.

Reply:
`$rootFile` --> `$rootFilePath`
`$rootInfo` --> `$rootHashes` and `$rootLength`
`$hash` --> `$trustedHash`
`$rootData` --> `$rootMetadata`

I trust this satisfies :)